### PR TITLE
Refactor: Use IFullPathResolver to resolve paths

### LINKED ISSUE #4316: Browse Diff Untracked: Delete and Edit menu items are not enabled
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
Functionality enabled in 2.51 #4031 is not enabled due to an incorrect check introduced in #4208 9f5fe413

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
 - View Unstaged commits
 - Edit and Delete menu items are not available

**Environment you encounter the issue:**

- GitExtensions version: master

**Did this work in previous version of GitExtensions (which)?**
Not in any released version

### PR DESCRIPTION
**Pull request by gerhardol**
Part of review for #4318, done in the occurrences where it seemed natural

Fixes #4316

Changes proposed in this pull request:
 - Consistently use IFullPathResolver to resolve paths to files
 
Screenshots before and after (if PR changes UI):
- None

What did I do to test the code and ensure quality:
 - Code review, partial tests

Has been tested on (remove any that don't apply):
 - Windows 10

### CODE REVIEW

File: GitUI/AutoCompletion/CommitAutoCompleteProvider.cs [hunk 1]
```diff
@@ -121,7 +121,8 @@ private static string GetChangedFileText (GitModule module, GitItemStatus file)
             // Try to read the contents of the file: if it cannot be read, skip the operation silently.
             try
             {
-                return File.ReadAllText(Path.Combine(module.WorkingDir, file.Name));
+                IFullPathResolver _fullPathResolver = new FullPathResolver(() => module.WorkingDir);
+                return File.ReadAllText(_fullPathResolver.Resolve(file.Name));
```
Comment (RussKie) [inline GitUI/AutoCompletion/CommitAutoCompleteProvider.cs]: This particular change doesn't feel like making the intent any clearer, nor the current implementation lends itself to a good refactor (I spent some time tinkering with it) without a major work.
Suggest rolling back the change here and let it remain in its current state. Eventually we may get around refactoring it.

File: GitUI/CommandsDialogs/FormGitAttributes.cs [hunk 2]
```diff
@@ -44,9 +46,9 @@ private void LoadFile()
         {
             try
             {
-                if (File.Exists(Module.WorkingDir + ".gitattributes"))
+                if (File.Exists(_fullPathResolver.Resolve(".gitattributes")))
                 {
-                    _NO_TRANSLATE_GitAttributesText.ViewFile(Module.WorkingDir + ".gitattributes");
+                    _NO_TRANSLATE_GitAttributesText.ViewFile(_fullPathResolver.Resolve(".gitattributes"));
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormGitAttributes.cs]: `Resolve` is called twice, please change it to call once

File: GitUI/CommandsDialogs/FormMailMap.cs [hunk 3]
```diff
@@ -45,9 +47,9 @@ private void LoadFile()
         {
             try
             {
-                if (File.Exists(Module.WorkingDir + ".mailmap"))
+                if (File.Exists(_fullPathResolver.Resolve(".mailmap")))
                 {
-                    _NO_TRANSLATE_MailMapText.ViewFile(Module.WorkingDir + ".mailmap");
+                    _NO_TRANSLATE_MailMapText.ViewFile(_fullPathResolver.Resolve(".mailmap"));
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormMailMap.cs]: `Resolve` is called twice, please change it to call once

File: GitUI/AutoCompletion/CommitAutoCompleteProvider.cs (hunk 1 above)
Comment (gerhardol) [reply GitUI/AutoCompletion/CommitAutoCompleteProvider.cs]: Reason to change was to set the pattern, but this was not obvious. Reverting.

**Pull request merged**